test/shared/src/main/scala/zio/test/TestClock.scala

```diff
@@ -294,12 +299,14 @@ object TestClock extends Serializable {
      * snapshot may not be fully consistent.
      */
     private def freeze(implicit trace: Trace): IO[Unit, Map[FiberId, Fiber.Status]] =
-      supervisedFibers.flatMap { fibers =>
-        ZIO.foldLeft(fibers)(Map.empty[FiberId, Fiber.Status]) { (map, fiber) =>
-          fiber.status.flatMap {
-            case done @ Fiber.Status.Done                    => ZIO.succeed(map.updated(fiber.id, done))
-            case suspended @ Fiber.Status.Suspended(_, _, _) => ZIO.succeed(map.updated(fiber.id, suspended))
-            case _                                           => ZIO.fail(())
+      freezeLock.withPermit {
+        supervisedFibers.flatMap { fibers =>
+          ZIO.foldLeft(fibers)(Map.empty[FiberId, Fiber.Status]) { (map, fiber) =>
+            fiber.status.flatMap {
+              case done @ Fiber.Status.Done                    => ZIO.succeed(map.updated(fiber.id, done))
+              case suspended @ Fiber.Status.Suspended(_, _, _) => ZIO.succeed(map.updated(fiber.id, suspended))
+              case _                                           => ZIO.fail(())
```

Comment:
```suggestion
              case _                                           => Exit.failUnit
```
?

Reply:
I'm tackling a lot of these in https://github.com/zio/zio/pull/9404, just need to iron out a few more things w.r.t the `TestClock` before I mark it as ready for review. Just wanted to have this PR separate since there is an open issue for it.

If that's okay, I'll leave it as is for this PR

Reply:
Sounds fine